frontend/src/pages/Task.tsx

```diff
@@ -0,0 +1,630 @@
+    name,
+    type,
+    onSave,
+    onDelete,
+}: {
+    name: string;
+    type: string;
+    onSave: (name: string, newName: string) => void;
+    onDelete: (name: string) => void;
+}) {
+    const [inputName, setInputName] = useState<string>(name);
+    const [isEditing, setIsEditing] = useState<boolean>(false);
+
+    const toggleEditing = () => {
+        if (!isEditing) {
+            setIsEditing(true);
+            return;
+        } else {
+            onSave(name, inputName);
+            setIsEditing(false);
+            return;
+        }
+    };
+
+    return (
+        <>
+            <div className=" flex flex-row items-center outline outline-2 outline-slate-200/80 p-2 rounded-md">
+                {!isEditing ? (
+                    <>
+                        <p className=" font-bold pr-2">{inputName}</p>
+                        <p className=" text-slate-200/80 mr-auto">{type}</p>
+                        <Button
+                            buttonIcon="edit"
+                            size="medium"
+                            style="minimal"
+                            outline={false}
+                            onAction={() => toggleEditing()}
+                        />
+                        <Button
+                            buttonIcon="trash"
+                            size="medium"
+                            style="minimal"
+                            outline={false}
+                            onAction={() => onDelete(inputName)}
+                        />
+                    </>
+                ) : (
+                    <>
+                        <input
+                            type="text"
+                            value={inputName}
+                            onChange={(evt) => setInputName(evt.target.value)}
+                            className=" mr-auto font-semibold text-gray-400 bg-dark-200 focus:ring-0 focus:outline-none shadow-none border-none pl-2 rounded-sm w-full"
+                        />
+                        <div className="ml-1" />
+                        <Button
+                            buttonIcon="tick"
+                            size="medium"
+                            style="bold"
+                            color="primary"
+                            outline={false}
+                            onAction={() => toggleEditing()}
+                        />
+                    </>
+                )}
+            </div>
+        </>
+    );
+}
+
+function TaskSidebarModel({ task }: { task: TaskInfo }) {
+    const { data, isLoading } = useGetModelsQuery();
+
+    const [isEditing, setIsEditing] = useState<boolean>(false);
+    const [selectedModel, setSelectedModel] = useState<string>("");
+    const [selectedModelVersion, setSelectedModelVersion] = useState<string>("");
+    const [updateTaskModel] = useUpdateTaskModelMutation();
+
+    const saveModel = () => {
+        const modelId = data?.models.find((m) => m.name == selectedModel)?.id;
```

Comment:
Use triple-equals here

```suggestion
        const modelId = data?.models.find((m) => m.name === selectedModel)?.id;
```